python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -1457,6 +1458,20 @@ def job_def(self) -> JobDefinition:
         """
         return self.op_execution_context.job_def
 
+    @public
+    def latest_materialization_event(
+        self, key: CoercibleToAssetKey
+    ) -> Optional[AssetMaterialization]:
+        """Get the most recent AssetMaterialization event for the key. Information like metadata and tags
+        can be found on the AssetMaterialization. If the key is not an upstream asset of the currently
+        materializing asset, None will be returned.
+
+        Returns: Optional[AssetMaterialization]
+        """
+        return self.op_execution_context._step_execution_context.upstream_asset_materialization_events.get(  # noqa: SLF001
+            AssetKey.from_coercible(key)
+        )
+
```

Comment:
My one concern is naming, in that it will perhaps lure the programmer into thinking that this will fetch the latest materialization event for _any_ asset, rather than only the immeadiate upstream.

My recommendation is

`latest_materialization_for_upstream_asset`

and then hard error with a clear error message if it is a key that is not immediately upstream.

I think the current behavior in this PR qualifies as "surprising" and will lead to subtle errors.